internal/painter/gl/gl_core.go

```diff
@@ -26,24 +27,30 @@ type Texture uint32
 // NoTexture is the zero value for a Texture
 var NoTexture = Texture(0)
 
-func newTexture() Texture {
+var textureFilterToGL = []int32{gl.LINEAR, gl.NEAREST}
+
+func newTexture(textureFilter canvas.ImageScale) Texture {
 	var texture uint32
 
+	if int(textureFilter) >= len(textureFilterToGL) {
+		panic(fmt.Sprintf("Invalid canvas.ImageScale value (%d)", textureFilter))
```

Comment:
As a library we should never panic.
I would recommend using fyne.LogError and then setting it to the default value and continuing

Reply:
I think this is a dev error and falling back to a default value can give the impression that any value is ok, which is not...

But is a philosophical thing, so if you want a log msg I can change.

Reply:
Is it the sort of developer error that is irrevocable and must force the application to close?
By leaving this in you are opening the possibility that a functioning app may force close in the middle of use. If we can fall back to a decent default we could save the user's sanity ;)

Reply:
something like:
```
	if int(textureFilter) >= len(textureFilterToGL) {
		fyne.LogError(fmt.Sprintf("Invalid canvas.ImageScale value (%d), using 0 (canvas.ImageScaleSmooth) as defualt value", textureFilter), nil)
		textureFilter = canvas.ImageScaleSmooth
	}
```

Reply:
Yes, except I would avoid `using 0 (canvas.ImageScaleSmooth)` and just say `using canvas.ImageScaleSmooth` - the specific int value is not something we should encourage.